### ISSUE TITLE
Add --title to Squirrel.exe pack?

### ISSUE THREAD

**Issue by RandomEngy** [clowd/Clowd.Squirrel#25]
The entry in Programs and Features is determined by the `--packName` parameter to `Squirrel.exe pack ...`

According to the [Squirrel naming conventions](https://github.com/Squirrel/Squirrel.Windows/blob/master/docs/using/naming.md#shortcut-name), that entry is set by the `title` property on the NuGet package metadata. Could we get a --title option added to `pack` to set this value when we are skipping the NuGet packaging? That way we could have spaces and dots in the Programs and Features entry.

**Comment by caesay** [comment]
Currently when using the pack command, both the <id> and <title> elements gets set to `--packName`. 

I have been thinking about we can solve this without adding a second parameter, for example.. "My App Name" passed in to `--packName` could get translated to "My.App.Name" or "MyAppName" for the <id> parameter, and the same logic could also be applied in the `UpdateManager` constructor.

Thoughts?

**Comment by RandomEngy** [comment]
Dots in the package ID [cause problems](https://github.com/Squirrel/Squirrel.Windows/blob/master/docs/using/naming.md#local-install-location): for example the install directory will strip out everything after the first one. I had tried using VidCoder.Beta as my packName and everything ended up being installed to `%localappdata%\VidCoder`.

In my particular case I had an old installer that would completely remove the `%localappdata%\VidCoder-Beta` folder on uninstall, so it was important for me to pick a different folder for Squirrel to install to. I ended up going with a pack name `VidCoder_Beta` so that running an old uninstaller wouldn't erase the install directory for the Squirrel installer.

Sanitizing the packName to create the `<id>` is probably a good "easy path" but avoiding adding a separate title would cause problems. For example if I released with a pack name of `VidCoder_Beta` I would not be able to take advantage of the new behavior because if I used `--packName "VidCoder Beta"` then it would create a package `<id>` of `VidCoderBeta` and I wouldn't be able to update cleanly.

**Comment by caesay** [comment]
Fixing the dots-in-the-package-name issue should be possible, I will have a look at that. If it's not possible or not easy to fix, then we should add a validation prohibiting package's with dots in the ID from being packaged.

Additionally, I would prefer that the default logic was that `packName` was used for the `<title>` parameter, and was be applied to `<id>` automatically as long as there are no invalid characters.

I can also create a `--packId` parameter which sets the `<id>` explicitly. If `--packId` is left blank, and there are any characters in `--packName` that are invalid for `<id>`, such as spaces, and maybe dots (depending on whether that can be fixed), the command line could throw an error and the message could inform you of the `--packId` parameter, and in this way there would be no hidden logic to translate "My App" to "MyApp", it would need to be done explicitly.

**Comment by caesay** [comment]
689b7ec778326cad5831eeb03ea441480ab99b29 I have fixed several issues related to using dots and dashes in package id's, `My-Package3.Cool` is a perfectly valid ID and is working fine now.

995dd6379a9a450948548285707567de7ace58e0 I have also deprecated `--packName` and replaced it with a separate id (required) and title (optional) parameter. `--packName` will continue to work the same way it used to, but it will not be listed in the `-h` output, and may be removed in a future version.

This will be in the next pre-release.
